packages/rule-components/src/RuleTable/RuleTable.js

```diff
@@ -0,0 +1,227 @@
+                filterKey: key
+            })
+        }));
+
+        return <section className="rhcs-c-rules-table">
+            <PrimaryToolbar
+                filterConfig={{ items: filterItems }}
+                activeFiltersConfig={{
+                    filters: Object.entries(filterValues).map(([ key, value ]) => calculateActiveFilters(
+                        filterItems.find(({ filterKey }) => filterKey === key),
+                        value,
+                        key
+                    )).filter(Boolean),
+                    onDelete: this.onFilterDelete
+                }}
+                pagination={pagination}
+                { ...toolbarProps }
+            />
+            <Table
+                {...props}
+                {...!isLoading && { actions } }
+                {...sortBy && { onSort: this.onSort }}
+                aria-label={ariaLabel}
+                cells={columns.map(({ transforms, ...column }) => ({
+                    ...column,
+                    transforms: [
+                        ...transforms || [],
+                        ...!isLoading && sortBy && !column.disableSort ? [ sortable ] : []
+                    ]
+                }))}
+                rows={
+                    isLoading ?
+                        [ ...new Array(5) ].map(() => ({
+                            cells: [{
+                                title: <Skeleton size="lg" />,
+                                props: { colSpan: columns.length + Boolean(actions) }
+                            }]
+                        })) :
+                        rows
+                }
+                {...!isLoading && detail && { onCollapse: this.onCollapse } }
+            >
+                <TableHeader />
+                <TableBody />
+            </Table>
+            <TableToolbar isFooter>
+                <Pagination
+                    {...pagination}
+                    widgetId="pagination-options-menu-bottom"
+                    variant={PaginationVariant.bottom}
+                    dropDirection="up"
+                />
+            </TableToolbar>
+        </section>;
+    }
+}
+
+RuleTable.propTypes = {
+    ariaLabel: PropTypes.string,
+    rules: PropTypes.shape({
+        meta: PropTypes.shape({
+            count: PropTypes.number,
+            offset: PropTypes.number,
+            limit: PropTypes.offset
+        }),
+        data: PropTypes.arrayOf(PropTypes.shape({
+            [PropTypes.string]: PropTypes.any
+        }))
+    }),
+    isLoading: PropTypes.bool,
+    columns: PropTypes.arrayOf(PropTypes.shape({
+        title: PropTypes.node,
+        selector: PropTypes.oneOfType([ PropTypes.string, PropTypes.arrayOf(PropTypes.string), PropTypes.func ]),
+        disableSort: PropTypes.bool
+    })),
+    sort: PropTypes.string,
+    defaultFilters: PropTypes.arrayOf(PropTypes.shape({})),
+    fetchData: PropTypes.func,
+    detail: PropTypes.oneOfType([ PropTypes.func, PropTypes.node ]),
```

Comment:
According to what I see in README it's should be a function, but it's not. So maybe here it should be `PropTypes.func`?

Reply:
It can be either function or react nore, I'll change it in README.

Reply:
Unfortunately it can’t, because it’s already used as function. But react node is text for example, or number, or anything that react can render. So I suggest to leave it only as function.